examples/hooks/components/InfiniteHits.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import { Hit as AlgoliaHit } from '@algolia/client-search';
+import {
+  useInfiniteHits,
+  UseInfiniteHitsProps,
+} from 'react-instantsearch-hooks';
+
+import { cx } from '../cx';
+
+export type InfiniteHitsProps<THit> = React.ComponentProps<'div'> &
+  UseInfiniteHitsProps & {
+    hitComponent: (props: { hit: THit }) => JSX.Element;
+  };
+
+export function InfiniteHits<THit extends AlgoliaHit<Record<string, unknown>>>({
+  hitComponent: Hit,
+  ...props
+}: InfiniteHitsProps<THit>) {
+  const { hits, isFirstPage, isLastPage, showMore, showPrevious } =
+    useInfiniteHits(props);
+
+  return (
+    <div className={cx('ais-InfiniteHits', props.className)}>
+      {props.showPrevious && (
+        <button
+          className={cx(
+            'ais-InfiniteHits-loadPrevious',
+            isFirstPage && 'ais-InfiniteHits-loadPrevious--disabled'
+          )}
+          onClick={showPrevious}
+          disabled={isFirstPage}
+        >
+          Show previous results
+        </button>
+      )}
+      <ol className="ais-InfiniteHits-list">
+        {hits.map((hit) => (
+          <li key={hit.objectID} className="ais-InfiniteHits-item">
+            <Hit hit={hit as unknown as THit} />
```

Comment:
we could avoid having this cast if useInfiniteHit<THit> passed down the generic to the hits render prop.
The connector is not doing that right now unfortunately. @Haroenv

Reply:
not sure how to do that properly, but we can introduce the generic in InstantSearch consistently afterwards, there's the same issue for hits and for autocomplete (more complicated as it can have multiple hits)